tests/phpunit/CRM/CiviAwards/Service/AwardImportTest.php

```diff
@@ -0,0 +1,161 @@
+ */
+class CRM_CiviAwards_Service_AwardImportTest extends BaseHeadlessTest {
+
+  /**
+   * Test the creation of the Case Type and the AwardDetail.
+   */
+  public function testImportAwardWithBaseInformation() {
+    $params = $this->getBaseParamsForAward();
+
+    (new AwardImportService())->create($params);
+
+    $awardCaseType = civicrm_api3('CaseType', 'getsingle', [
+      'title' => $params['title'],
+    ]);
+
+    $awardDetail = civicrm_api3('AwardDetail', 'getsingle', [
+      'case_type_id' => $awardCaseType['name'],
+    ]);
+    $this->assertCaseTypeInformation($params, $awardCaseType);
+    $this->assertDetailsInformation($params, $awardDetail);
+  }
+
+  /**
+   * Test the Award creation with manager information.
+   *
+   * In this test only one manager is specified.
+   */
+  public function testImportAwardWithSingleManagerInformation() {
+    $contactId = CRM_CiviAwards_Test_Fabricator_Contact::fabricate()['id'];
+    $params = array_merge(
+      $this->getBaseParamsForAward(),
+      [
+        'award_manager' => $contactId,
+      ]
+    );
+
+    (new AwardImportService())->create($params);
+
+    $awardCaseType = civicrm_api3('CaseType', 'getsingle', [
+      'title' => $params['title'],
+    ]);
+    $awardDetail = civicrm_api3('AwardDetail', 'getsingle', [
+      'case_type_id' => $awardCaseType['name'],
+    ]);
+
+    $this->assertCaseTypeInformation($params, $awardCaseType);
+    $this->assertDetailsInformation($params, $awardDetail);
+    $this->assertEquals($contactId, $awardDetail['award_manager'][0]);
+  }
+
+  /**
+   * Test the Award creation with manager information.
+   *
+   * In this test two managers are specified.
+   */
+  public function testImportAwardWithMultipleManagerInformation() {
+    $contactIdOne = CRM_CiviAwards_Test_Fabricator_Contact::fabricate()['id'];
+    $contactIdTwo = CRM_CiviAwards_Test_Fabricator_Contact::fabricate()['id'];
+    $params = array_merge(
+      $this->getBaseParamsForAward(),
+      [
+        'award_manager' => "$contactIdOne,$contactIdTwo",
+      ]
+    );
+
+    (new AwardImportService())->create($params);
+
+    $awardCaseType = civicrm_api3('CaseType', 'getsingle', [
+      'title' => $params['title'],
+    ]);
+    $awardDetail = civicrm_api3('AwardDetail', 'getsingle', [
+      'case_type_id' => $awardCaseType['name'],
+    ]);
+
+    $this->assertCaseTypeInformation($params, $awardCaseType);
+    $this->assertDetailsInformation($params, $awardDetail);
+    $this->assertCount(2, $awardDetail['award_manager']);
+    $this->assertContains($contactIdOne, $awardDetail['award_manager']);
+    $this->assertContains($contactIdTwo, $awardDetail['award_manager']);
+  }
+
+  /**
+   * Test that in case of error, all the operation is cancelled.
+   */
+  public function testErrorImportingAwardWithDetailInformation() {
+    $params = $this->getBaseParamsForAward();
+    $params['start_date'] = 'invalid date';
+    $this->expectException(API_Exception::class);
```

Comment:
Done: https://github.com/compucorp/uk.co.compucorp.civiawards/pull/164/commits/175f1e4a98048385990c5e4f10d51b36ea9d1be3
(I was tempted to check with a regular expression the presence of some words that implies the error expected, but I think that this will be good enough)